### DISCUSSION TITLE
how to truncate a collection?

### DISCUSSION THREAD

**Discussion by AlizerUncaged** [pocketbase/pocketbase#2950] [Q&A] [answered]
I have a massive collection of around 50,000 records and now I need to clear it, how do I do that?

**Comment by ganigeorgiev** [comment, marked as answer]
We don't have dedicated web API for this currently (see https://github.com/pocketbase/pocketbase/issues/2238).

To delete all collection entries, you could:

1. Use the sqlite3 cli tool and run something like `sqlite3 ./pb_data/data.db 'DELETE FROM YOUR_COLLECTION'`

2. Duplicate the existing collection (_Admin UI > Collections > Edit collection (settings cogwheel) > 3-dots menu > Duplicate_), delete the old collection and then rename the new collection back to the original.

**Comment by AlizerUncaged** [comment]
alright i see, seem like i need to edit the sqlite file manually

**Comment by wolfsolver** [comment]
Just as reference (since I search for the same thing an reach this), is possible to truncate via admin console.

**Reply by ganigeorgiev** [reply]
Yes, from the specific collection settings dropdown:

![collection_dropdown](https://i.imgur.com/vxs8Ls6.png)
